mobile_verifier/src/reward_shares.rs

```diff
@@ -197,6 +204,112 @@ impl MapperShares {
     }
 }
 
+#[derive(Default)]
+pub struct ServiceProviderShares {
+    pub shares: Vec<ServiceProviderDataSession>,
+}
+
+impl ServiceProviderShares {
+    pub fn new(shares: Vec<ServiceProviderDataSession>) -> Self {
+        Self { shares }
+    }
+
+    pub async fn from_payers_dc(
+        payer_shares: HashMap<String, u64>,
+        client: &dyn CarrierServiceVerifier<Error = ClientError>,
+    ) -> anyhow::Result<ServiceProviderShares> {
+        let mut shares = vec![];
+        for (payer, total_dcs) in payer_shares {
+            let service_provider = Self::payer_key_to_service_provider(&payer, client).await?;
+            shares.push(ServiceProviderDataSession {
+                service_provider,
+                total_dcs: Decimal::from(total_dcs),
+            })
+        }
+        Ok(ServiceProviderShares { shares })
+    }
+
+    fn total_dc(&self) -> Decimal {
+        self.shares.iter().map(|v| v.total_dcs).sum()
+    }
+
+    pub fn get_scheduled_tokens_for_service_providers(&self, duration: Duration) -> Decimal {
```

Comment:
should this be a method off the `ServiceProviderShares` struct or a free function alongside `get_total_scheduled_tokens`?

Reply:
these getters are all moved to free functions in #683 which will go live with service provider rewards